responses/11-targets-debug.md

```diff
@@ -6,17 +6,21 @@ If you've written your own functions or scripts before, you may have run into th
 
 Breakpoints allow you to run a function (or script) up until the line of the breakpoint, and then the evaluation pauses. You are able to inspect all variables available at that point in the evaluation, and even step carefully forward one line at a time. It is out of scope of this exercise to go through exactly how to use debuggers, but they are powerful and helpful tools. It would be a good idea to read up on them if you haven't run into breakpoints yet. 
 
-In `scipiper`, you can't set a breakpoint in the "normal" way, which would be clicking on the line number after you sourced the script. Instead, you need to use the other method for debugging in R, which requires adding the function call `browser()` to the line where you'd like the function call to stop. 
+In `targets`, you can't set a breakpoint in the "normal" way, which would be clicking on the line number after you sourced the script. Instead, you need to use the other method for debugging in R, which requires adding the function call `browser()` to the line where you'd like the function call to stop and specifying an additional argument when you call `tar_make()`. 
 
 ---
 
-You have a working, albeit brittle, pipeline in your course repository. You can try it out with `scipiper::scmake()`. This pipeline has a number of things you'll work to fix later, but for now, it is a useful reference. The pipeline contains several functions which are defined in `.R` files. 
+You have a working, albeit brittle, pipeline in your course repository. You can try it out with `targets::tar_make()`. This pipeline has a number of things you'll work to fix later, but for now, it is a useful reference. The pipeline contains several functions which are defined in `.R` files. 
 
-So, if you wanted to look at what `download_files` where created within the `download_nwis_data` function, you could set a breakpoint by adding `browser()` to the `"1_fetch/src/get_nwis_data.R"` file:
+So, if you wanted to look at what `download_files` where created within the `download_nwis_data` function, you could set a breakpoint by adding `browser()` to the `"1_fetch/src/get_nwis_data.R"` file. Hint: to quickly navigate to this function source code from your makefile, you can put your cursor on the name of the function then click F2 and it will take you to the correct location in the corresponding source file!
 
 ![browser()](https://user-images.githubusercontent.com/2349007/82158816-bed9bb00-984f-11ea-8892-b2aeb5e4818d.png)
 
-Then, running `scmake()` will land you right in the middle of line 8. Give it a try on your own.
+There is one more step to get your breakpoint to work in `targets`. You will need to add `callr_function = NULL` to your `tar_make()` call. When you run `tar_make(callr_function = NULL)`, you will land right in the middle of line 8. Give it a try on your own.
```

Comment:
Hmmm I don't remember reading that and when I was trying this debugging stuff out, I definitely didn't restart R before or after my debugging tests. Let me do a few tests and see what happens.

Reply:
I am still unimpacted by this, but from searching through the documentation, it looks like this may be if you are setting `debug` in the `tar_option_set()` function: https://github.com/ropensci/targets/blob/7a76cf8c94145f2f4b70f0859dba7eb5d918396c/R/tar_option_set.R#L60-L61

Though, I am not sure when you need to do that vs adding `browser()` to a particular function and running `tar_make(callr_function = NULL)`.

Reply:
I am going to skip mentioning since I have yet to need to follow these guidelines.